### ISSUE TITLE
`kfp-profile-controller` integration test fails with `ApiError: configmaps "metadata-grpc-configmap" not found`

### ISSUE THREAD

**Issue by NohaIhab** [canonical/kfp-operators#368]
### Bug Description

the `test_sync_webhook` integration test for `kfp-api` charm is failing consistently with the log:
```
httpx.HTTPStatusError: Client error '404 Not Found' for url 'https://10.1.0.33:16443/api/v1/namespaces/profilename/configmaps/metadata-grpc-configmap'
```
seen in [this run of the CI](https://github.com/canonical/kfp-operators/actions/runs/6669069136/job/18126036351#step:5:181)

### To Reproduce

run the integration tests for `kfp-profile-controller` charm

### Environment

microk8s 1.25-strict/stable
juju 3.1/stable

### Relevant log output

```shell
=================================== FAILURES ===================================
______________________________ test_sync_webhook _______________________________
Traceback (most recent call last):
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.8/site-packages/lightkube/core/generic_client.py", line 188, in raise_for_status
    resp.raise_for_status()
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.8/site-packages/httpx/_models.py", line 749, in raise_for_status
    raise HTTPStatusError(message, request=request, response=self)
httpx.HTTPStatusError: Client error '404 Not Found' for url 'https://10.1.0.33:16443/api/v1/namespaces/profilename/configmaps/metadata-grpc-configmap'
For more information check: https://httpstatuses.com/404

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.8/site-packages/_pytest/runner.py", line 341, in from_call
    result: Optional[TResult] = func()
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.8/site-packages/_pytest/runner.py", line 262, in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.8/site-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.8/site-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.8/site-packages/pluggy/_callers.py", line 152, in _multicall
    return outcome.get_result()
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.8/site-packages/pluggy/_result.py", line 114, in get_result
    raise exc.with_traceback(exc.__traceback__)
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.8/site-packages/pluggy/_callers.py", line 77, in _multicall
    res = hook_impl.function(*args)
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.8/site-packages/_pytest/runner.py", line 177, in pytest_runtest_call
    raise e
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.8/site-packages/_pytest/runner.py", line 169, in pytest_runtest_call
    item.runtest()
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.8/site-packages/_pytest/python.py", line 1792, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.8/site-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.8/site-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.8/site-packages/pluggy/_callers.py", line 152, in _multicall
    return outcome.get_result()
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.8/site-packages/pluggy/_result.py", line 114, in get_result
    raise exc.with_traceback(exc.__traceback__)
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.8/site-packages/pluggy/_callers.py", line 77, in _multicall
    res = hook_impl.function(*args)
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.8/site-packages/_pytest/python.py", line 194, in pytest_pyfunc_call
    result = testfunction(**testargs)
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.8/site-packages/pytest_asyncio/plugin.py", line 532, in inner
    _loop.run_until_complete(task)
  File "/usr/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/home/runner/work/kfp-operators/kfp-operators/charms/kfp-profile-controller/tests/integration/test_charm.py", line 218, in test_sync_webhook
    lightkube_client.get(resource, name=name, namespace=profile)
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.8/site-packages/lightkube/core/client.py", line 140, in get
    return self._client.request("get", res=res, name=name, namespace=namespace)
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.8/site-packages/lightkube/core/generic_client.py", line 245, in request
    return self.handle_response(method, resp, br)
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.8/site-packages/lightkube/core/generic_client.py", line 196, in handle_response
    self.raise_for_status(resp)
  File "/home/runner/work/kfp-operators/kfp-operators/.tox/integration/lib/python3.8/site-packages/lightkube/core/generic_client.py", line 190, in raise_for_status
    raise transform_exception(e)
lightkube.core.exceptions.ApiError: configmaps "metadata-grpc-configmap" not found
```

**Comment by NohaIhab** [comment]
the `test_sync_webhook` integration test creates a profile using the fixture:
https://github.com/canonical/kfp-operators/blob/17cd94261dfc7a1c5b61e861319ce2a0334252a5/charms/kfp-profile-controller/tests/integration/test_charm.py#L101-L117

and then checks for the resources created in that profile's namespace
it seems that this is a race where the resources are just not created yet when we check for them, since the test is using `lightkube.get` without waiting on those resources.

the configmap `metadata-grpc-configmap` just happens to be the first resource in the list that we check on, the test simply fails to get the first resource and aborts there.

**Comment by NohaIhab** [comment]
After inspecting the cluster when tests are finished, the resources for the profile never get created, and we can see logs in the metacontroller charm:
```
{"level":"error","ts":1698933302.552061,"logger":"controller.decorator-metacontroller","msg":"Reconciler error","name":"kubeflow-pipelines-profile-controller","namespace":"","error":"can't find child resource \"destinationrules\" in networking.istio.io/v1alpha3","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.2/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.2/pkg/internal/controller/controller.go:227"}
```
metacontroller cannot find the CRD for destinationrules.
The issue here is that the `DecoratorController` was incorrectly modified in the 1.8 updates.
See issue #373 , that's expected to be the root cause of the failing test.

**Comment by NohaIhab** [comment]
closed by #376
